src/internet/digitally/digitallyimportedservicebase.cpp

```diff
@@ -181,11 +183,59 @@ void DigitallyImportedServiceBase::ShowContextMenu(const QPoint& global_pos) {
     context_menu_->addAction(IconLoader::Load("configure", IconLoader::Base),
                              tr("Configure..."), this,
                              SLOT(ShowSettingsDialog()));
+    context_menu_->addAction(GetCopySelectedPlayableItemURLAction());
+  }
+}
+
+void DigitallyImportedServiceBase::ShowContextMenu(const QPoint& global_pos) {
+  EnsureMenuCreated();
+  QStandardItem* item = model()->itemFromIndex(model()->current_index());
+
+  if (item) {
+    int type = item->data(InternetModel::Role_Type).toInt();
+
+    // Digitally Imported streams have a Role_Type of 0.
+    if (type == 0) {
```

Comment:
This conditional could be tidied up by replacing with `bool can_play`.

This bit of code was successfully tested **before** a premium account became mandatory, so any changes will need to be tested by someone with such an account.